feat: Add support for sentinel writer

mhewedy opened the pull request:
We have a case where the target is using sentinel.

And manually configuring the master address in redis-shake is cumbersome because the Redis master changes anytime due to failover to a new master node.

mhewedy committed:
add support for sentinel writer

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/tair-opensource/RedisShake?pullRequest=806) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/tair-opensource/RedisShake?pullRequest=806) before we can accept your contribution.<br/><hr/>**mhewedy** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/tair-opensource/RedisShake?pullRequest=806) it.</sub>

suxb201 commented:
@mhewedy LGTM, but please sign the CLA so we can merge your changes.

mhewedy commented:
Thanks for your feedback

i already singed the CLA, but seems there’s some issue preventing it from reflecting here (see blow)

Any suggestions?

![image](https://github.com/tair-opensource/RedisShake/assets/1086049/acf7d59a-b5fb-466c-ac27-cb7ef7ee2d5d)

mhewedy committed:
parent 72a34aa642d09e2b58d0f2d7edb762a8e7dabd73
author mhewedy <mhewedy@gmail.com> 1716438272 +0300
committer mhewedy <mhewedy@gmail.com> 1716438342 +0300

add support for sentinel writer

mhewedy commented:
> @mhewedy LGTM, but please sign the CLA so we can merge your changes.

I've signed the CLA now.

mhewedy commented:
@suxb201 

Thank you for merging my recent PR. I’m excited to see the changes included in the next release. 
Could you please let me know if there is an expected release date for the upcoming version?

suxb201 commented:
@mhewedy v4.1.0 has been released: https://github.com/tair-opensource/RedisShake/releases/tag/v4.1.0
